### PR TITLE
Email verification

### PR DESCRIPTION
**Pull request by panel14**

### CODE REVIEW

**Comment by IP-13** [comment]
|Overall Project|65.47% **`-2.16%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|3.97%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/fa70ecaf73218260203b983ddcb1f8d8ed347eea/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|76.24% **`-2.48%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/fa70ecaf73218260203b983ddcb1f8d8ed347eea/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|20% **`-32.86%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/fa70ecaf73218260203b983ddcb1f8d8ed347eea/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|17.86% **`-24.4%`**|:x:|
|[UserEmailVerificationInfoService.java](https://github.com/itmo-events-app/event-app/blob/fa70ecaf73218260203b983ddcb1f8d8ed347eea/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserEmailVerificationInfoService.java)|14.63% **`-85.37%`**|:x:|

File: src/main/resources/db/migration/V0_2__create_tables.sql [hunk 1]
```diff
@@ -123,6 +123,13 @@ create table if not exists task
     reminder timestamp not null
 );
 
+create table if not exists user_email_verification_info(
```
Comment (IP-13) [inline src/main/resources/db/migration/V0_2__create_tables.sql:126]: Надо в тестах в миграцию добавить тоже

File: src/main/java/org/itmo/eventapp/main/controller/AuthController.java [hunk 2]
```diff
@@ -61,4 +57,22 @@ ResponseEntity<List<RegistrationRequestForAdmin>> listRegisterRequests() {
                 .status(HttpStatus.OK)
                 .body(authenticationService.listRegisterRequestsCallback());
     }
+
+    @PostMapping("/sendVerificationEmail")
+    ResponseEntity<Void> sendVerificationEmail(@RequestParam String returnUrl) {
+        authenticationService.sendVerificationEmail(returnUrl);
+        return ResponseEntity.status(HttpStatus.OK).build();
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/controller/AuthController.java:64]: 204 статус

**Comment by IP-13** [comment]
|Overall Project|63.92% **`-2.06%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|4%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/14b457088b52a788f683af4ae8d2bcfa83fac696/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|76.24% **`-2.48%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/14b457088b52a788f683af4ae8d2bcfa83fac696/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|20% **`-32.86%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/14b457088b52a788f683af4ae8d2bcfa83fac696/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|17.75% **`-23.96%`**|:x:|
|[UserEmailVerificationInfoService.java](https://github.com/itmo-events-app/event-app/blob/14b457088b52a788f683af4ae8d2bcfa83fac696/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserEmailVerificationInfoService.java)|14.63% **`-85.37%`**|:x:|

File: src/main/java/org/itmo/eventapp/main/controller/AuthController.java [hunk 3]
```diff
@@ -56,4 +57,22 @@ ResponseEntity<List<RegistrationRequestForAdmin>> listRegisterRequests() {
...
+    @PostMapping("/sendVerificationEmail")
+    ResponseEntity<Void> sendVerificationEmail(@RequestParam String returnUrl) {
+        authenticationService.sendVerificationEmail(returnUrl);
+        return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+    }
+
+    @PostMapping("/validateEmailVerificationToken")
+    ResponseEntity<Void> validateEmailVerificationToken(@RequestParam String token) {
+        authenticationService.validateEmailVerificationToken(token);
+        return ResponseEntity.status(HttpStatus.OK).build();
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/controller/AuthController.java:70]: здесь тоже надо 204 статус, и в следующем методе тоже

**Comment by IP-13** [comment]
|Overall Project|64.08% **`-2.06%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|4%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/be95c30bf50f2a4bb41e741b7c6d685d5091ae48/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|76.24% **`-2.48%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/be95c30bf50f2a4bb41e741b7c6d685d5091ae48/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|20% **`-32.86%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/be95c30bf50f2a4bb41e741b7c6d685d5091ae48/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|17.75% **`-23.96%`**|:x:|
|[UserEmailVerificationInfoService.java](https://github.com/itmo-events-app/event-app/blob/be95c30bf50f2a4bb41e741b7c6d685d5091ae48/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserEmailVerificationInfoService.java)|14.63% **`-85.37%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|63.92% **`-2.06%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|4%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/1637c34252fd87d351f44f69ad8552cb1b52754c/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|76.24% **`-2.48%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/1637c34252fd87d351f44f69ad8552cb1b52754c/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|20% **`-32.86%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/1637c34252fd87d351f44f69ad8552cb1b52754c/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|17.75% **`-23.96%`**|:x:|
|[UserEmailVerificationInfoService.java](https://github.com/itmo-events-app/event-app/blob/1637c34252fd87d351f44f69ad8552cb1b52754c/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserEmailVerificationInfoService.java)|14.63% **`-85.37%`**|:x:|

**Review by omgronny — APPROVED** [review]

**Pull request merged**